File: backend/product-service/src/main/java/com/lexo/productservice/entity/Product.java
package com.lexo.productservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 2000)
    private String description;

    @ElementCollection
    @CollectionTable(name = "product_bullets", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "bullet", length = 2000)
    private List<String> bullets;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "product_url")
    private String productUrl;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @ElementCollection
    @CollectionTable(name = "product_tags", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "tag", length = 500)
    private List<String> tags;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/product.ts
export interface Product {
  id: number;
  title: string;
  description?: string;
  bullets?: string[];
  imageUrl?: string;
  productUrl?: string;
  price: number;
  tags?: string[];
  createdAt: string;
  updatedAt?: string;
}

export interface CreateProductDto {
  title: string;
  description?: string;
  bullets?: string[];
  imageUrl?: string;
  productUrl?: string;
  price: number;
  tags?: string[];
}

export interface UpdateProductDto extends Partial<CreateProductDto> {}

export interface PresignedUrlResponse {
  presignedUrl: string;
}

// Lite version for pickers/dropdowns
export interface ProductLite {
  id: number;
  title: string;
  imageUrl?: string;
  productUrl?: string;
  price?: number;
}

